src/tests/unit/credential_providers/test_aws_secrets_manager.py

```diff
@@ -0,0 +1,56 @@
+# *******************************************************
+# Copyright (c) VMware, Inc. 2020-2021. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""Tests for the AWSCredentialProvider"""
+
+from unittest import mock
```

Comment:
I'm hesitant adding yet another mock framework. Is this not possible with monkeypatch?

Reply:
Good point, Alex (@avanbrunt-cb), I was also interested how this could be done and tested with a few things - with mox it could be done and the tests can look something like:
`def test_raising_exception_on_emtpy_secret(mox):
    """Test for raising the CredentialError."""
    mox.StubOutWithMock(boto3.session.Session, 'client')
    client_mock = mox.CreateMockAnything()
    client_mock.get_secret_value(SecretId='test').AndReturn({})
    instance = boto3.session.Session.__new__(boto3.session.Session)
    instance.client(region_name='us-east-2', service_name='secretsmanager').AndReturn(client_mock)

    mox.ReplayAll()
    with pytest.raises(CredentialError):
        AWSCredentialProvider(secret_arn="test").get_credentials()
    mox.VerifyAll()


def test_aws_getting_credentials(mox):
    """Test for getting credentials."""
    test_data = {
        "url": "<URL>",
        "token": "<TOKEN>",
        "org_key": "<ORG_KEY>",
        "ssl_verify": True,
        "ssl_verify_hostname": True,
        "ssl_cert_file": "<FILE_PATH>",
        "ssl_force_tls_1_2": True,
        "proxy": "<NAME_OF_THE_PROXY_HOST>",
        "ignore_system_proxy": True,
        "integration": "<INTEGRATION_NAME>"
    }
    mox.StubOutWithMock(boto3.session.Session, 'client')
    client_mock = mox.CreateMockAnything()
    client_mock.get_secret_value(SecretId='test').AndReturn({"SecretString": json.dumps(test_data)})
    instance = boto3.session.Session.__new__(boto3.session.Session)
    instance.client(region_name='us-east-2', service_name='secretsmanager').AndReturn(client_mock)

    mox.ReplayAll()
    credentials = AWSCredentialProvider(secret_arn="test").get_credentials()
    assert credentials.to_dict() == test_data
    mox.VerifyAll()`

And of course we will need to `import mox`

Reply:
I am hesitant to use pymox as it is not being maintained. There have been no updates since 2018 and is one of the reasons we have a warning at the moment.

Reply:
I think we should prioritize the python builtins first and then the 3rd party libs that we use. I find the built-in mock lib perfectly capable of doing something like that so I've used it instead of a `pymox`.

Reply:
We are pytest as our testing framework which has monkeypatch built-in. I would prefer to reduce the number of mocking frameworks we use unless you feel monkeypatch would but be sufficient here

Reply:
Noted, I changed it with monkeypatch.